### ISSUE TITLE
.pulp-cache should consider the output directory

### ISSUE THREAD

**Issue by natefaubion** [purescript-contrib/pulp#146]
I tried trashing my output directory to trigger a clean compile, but pulp wouldn't rebuild because its only considering source files.

Additionally a `pulp clean` command might be nice as a convenience.

**Comment by maddie927** [comment]
A `pulp clean` command would be great!  The cache should also be aware of command differences, since `pulp build/browserify` and `pulp server` compile code into `./output` differently.  You can't `pulp browserify -O` after running `pulp server`, but you can `pulp server` then `rm -rf .pulp-cache output && pulp browserify -O`.

**Comment by hdgarrood** [comment]
the `pulp server` thing needs addressing separately, IMO: #151

**Comment by hdgarrood** [comment]
What would `pulp clean` do? Just `rm -r output`?

**Comment by ttbodil** [comment]
`pulp --force` kills `.pulp-cache`, isn't that sufficient?

On the other hand, I'm starting to wonder if `.pulp-cache` was such a good idea in the first place. The only real gain is in being able to skip `psc-bundle`, `psc` is able to do incremental builds just fine on its own. There seem to be so many edge cases to consider that it might be better to just throw the whole mechanism out at this point.

Now, if only `psc` were able to actually cleanly signal if it recompiled anything... or we could scan its output for "Compiling" messages, I suppose, but that's quite brittle.

**Comment by hdgarrood** [comment]
If you're doing `pulp browserify -O`, you can skip `browserify` as well, not just `psc-bundle`, and both of these will probably take a while between them. But yeah, I'm inclined to agree.

**Comment by maddie927** [comment]
@ttbodil I thought `--force` would fix it, but it doesn't seem to actually rebuild:

```
pulp server
pulp browserify -O --force
-- errors --
rm -rf .pulp-cache output
pulp browserify -O
-- works --
```

**Comment by maddie927** [comment]
But I'm also inclined to agree that the cache doesn't help a lot.  Speed matters when developing so you can get quick feedback from your editor, but it doesn't matter much for a full release build (the only time I'd use `pulp browserify`).  I usually run clean scripts before doing release builds anyway.

**Comment by hdgarrood** [comment]
Oh that's interesting/unfortunate. If we wanted to address that, we'd have to either modify `psc` to force rebuilds whenever the `require-path` option changes, or get pulp to remove the build path at the start of `--force` builds.

**Comment by cdepillabout** [comment]
I ran into the same problem as @natefaubion.  I nuked my `output/` directory hoping to trigger a complete rebuild next time I rant `pulp build`, but it wouldn't rebuild.

It would be nice if `pulp build` was idempotent, so I could expect the output files (the stuff under `output/`) to be the same as long as the input files were the same (the stuff under `src/`).

**Comment by hdgarrood** [comment]
Resolved by #165.
